c/termination-numeric-2/lychrel_false-termination.c

```diff
@@ -0,0 +1,35 @@
+extern int __VERIFIER_nondet_int(void);
+
+long number, reversedNr = 0;
+
+long reverse() {
+  long remainder;
+  reversedNr = 0;
+
+  while (number > 0) {
+    remainder = number % 10;
+    reversedNr = (reversedNr * 10) + remainder;
+    number /= 10;
+  }
+  return reversedNr;
+}
+
+long isPalindrome() { return number == reverse(); }
+
+int isLychrel() {
+  while (1) {
+    if (isPalindrome()) {
+      return 0;
+    } else {
+      number = reversedNr + number;
```

Comment:
This will eventually lead to a (signed) long overflow, e.g. for an initial value of `196` for `number`. This would then be undefined behavior,  so the termination label can not be added. Replacing the `long` types by `unsigned long` would fix this as unsigned overflows are defined behavior in the C11 standard.
The loop iteration limit in the task `lychrel_true-termination.c` might prevent this overflow, but I would replace the `long` types there too just to be sure.